version: '3.8'
services:
  app:
    build:
      context: ./app
    # No direct port mapping; access is via Nginx reverse proxy
    depends_on:
      - db
      - gitea
    environment:
      MYSQL_HOST: db
      MYSQL_DATABASE: vulnerable_db
      MYSQL_USER: vuln_user
      MYSQL_PASSWORD: vuln_pass
      GITEA_URL: http://gitea:3000
  db:
    image: mysql:8.0
    restart: always
    environment:
      MYSQL_ROOT_PASSWORD: rootpass
      MYSQL_DATABASE: vulnerable_db
      MYSQL_USER: vuln_user
      MYSQL_PASSWORD: vuln_pass
    volumes:
      - db_data:/var/lib/mysql
      - ./db/init.sql:/docker-entrypoint-initdb.d/init.sql
  nginx:
    image: nginx:1.27
    ports:
      - "80:80"
    depends_on:
      - app
    volumes:
      - ./nginx/conf.d/default.conf:/etc/nginx/conf.d/default.conf
  gitea:
    image: gitea/gitea:1.12.5
    ports:
      - "3000:3000"
    environment:
      USER_UID: "1000"
      USER_GID: "1000"
      GITEA__database__DB_TYPE: mysql
      GITEA__database__HOST: db:3306
      GITEA__database__NAME: vulnerable_db
      GITEA__database__USER: vuln_user
      GITEA__database__PASSWD: vuln_pass
    volumes:
      - gitea_data:/data
    depends_on:
      - db
volumes:
  db_data:
  gitea_data: